### PR TITLE
Ft add results

### PR DESCRIPTION
**Pull request by jamsgra**
#### What does this PR do?
- update admin `user` with `school_uid` when school created
- add `school_uid` to `req.session` when logging in
- add `school_uid` to `addUsers` data **(issue #14)**
- create `resultsController`
- `addResults` to `Results` api route
- `addResultsValidation`
- `excelValidators` helper functions
- fixes bugs in database
- fixes `schoolController` tests

#### Description of Task to be completed?

- update session with `school_uid`
-  create `resultsController` with `addResults` function
- add middleware & helper functions to validate excel data
- test `addResultsValidation` middleware
- test `addResults`

#### How should this be manually tested?

- `npm i`
- `npm run db-init-test`
- `npm test`

#### Any background context you want to provide?

- Addresses issue #14

### CODE REVIEW

File: server/models/results.js [hunk 1]
```diff
@@ -4,7 +4,8 @@ export default (sequelize, DataTypes) => {
     result_id: {
       allowNull: false,
       primaryKey: true,
-      type: DataTypes.INTEGER
+      type: DataTypes.INTEGER,
+      defaultValue: 0,
```
Comment (sulenchy) [inline server/models/results.js]: Can we set the defaultValue to 1? I think it sounds better than having it to be zero.

File: tests/server/middleware/addResultsValidation.test.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        })
+      }).catch(err => console.log(err))
+  });
+
+  // before((done) => {
+  //   agent
+  //       .post(createSchoolUrl)
+  //       .send(schoolDataWithValidFields)
+  //       .end((err, res) => {
+  //           const { school } = res.body;
```
Comment (sulenchy) [inline tests/server/middleware/addResultsValidation.test.js:70]: We can also get it rid of commented code @jamsgra

**Comment by jamsgra** [comment]
@sulenchy I realised there were some bugs with tests (and then realised I had to fix a couple of things) along with resolving issue #14. Let me know what you think.

**Review by sulenchy — APPROVED** [review]
Looks good to me

**Comment by jamsgra** [comment]
@sulenchy I've pushed some new changes:

- Results table: add `student_result_id` column
- Add `student_result_id` to results
- Check for duplicates in results file
- Check for duplicates in results table
- Tests

Let me know what you think.

File: server/helpers/convertIndexToExcelRow.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const convertIndexToExcelRow = (obj) => {
```
Comment (sulenchy) [inline server/helpers/convertIndexToExcelRow.js:1]: @jamsgra I think jsdoc is necessary here

File: server/helpers/sendError.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const sendError = (res, statusCode, error) => {
+  return res.status(statusCode).json({
```
Comment (sulenchy) [inline server/helpers/sendError.js:2]: Thanks for this abstraction @jamsgra

**Review by sulenchy — APPROVED** [review]
Good job man.

**Pull request merged**